File: app/src/main/java/it/amonshore/comikkua/LogHelper.java
package it.amonshore.comikkua;

import android.util.Log;

import androidx.annotation.NonNull;

public class LogHelper {

    private static String mTag = "CMK";

    public static void setTag(@NonNull String tag) {
        mTag = tag;
    }

    public static void d(String message) {
        Log.d(mTag, message);
    }

    public static void d(String message, Object... args) {
        Log.d(mTag, String.format(message, args));
    }

    public static void w(String message) {
        Log.w(mTag, message);
    }

    public static void w(String message, Object... args) {
        Log.w(mTag, String.format(message, args));
    }

    public static void e(String message) {
        Log.e(mTag, message);
    }

    public static void e(String message, Exception error) {
        Log.e(mTag, message, error);
    }

    public static void e(Exception error, String message, Object... args) {
        Log.e(mTag, String.format(message, args), error);
    }

    public static void e(String message, Object... args) {
        Log.e(mTag, String.format(message, args));
    }

}

File: app/src/main/java/it/amonshore/comikkua/LogHelper.kt
package it.amonshore.comikkua

import android.util.Log

object LogHelper {

    private const val TAG = "CMK"

    @Deprecated("Usare versione con supplier")
    @JvmStatic
    fun d(message: String) {
        if (BuildConfig.DEBUG) {
            Log.d(TAG, message)
        }
    }

    @JvmStatic
    fun d(supplier: () -> String) {
        if (BuildConfig.DEBUG) {
            Log.d(TAG, supplier())
        }
    }

    @JvmStatic
    fun i(message: String) {
        Log.i(TAG, message)
    }

    @JvmStatic
    fun w(message: String) {
        Log.w(TAG, message)
    }

    @JvmStatic
    fun e(message: String) {
        Log.e(TAG, message)
    }

    @JvmStatic
    fun e(message: String, error: Throwable? = null) {
        Log.e(TAG, message, error)
    }

    @JvmStatic
    fun e(error: Throwable) {
        Log.e(TAG, null, error)
    }
}